### PR TITLE
Hide previews for published posts on Jetpack sites

### LINKED ISSUE #10161: Show an error for Remote Previews on Jetpack sites
Due to an issue in the API with Jetpack sites, generated previews are not up to date.

Until the API is fixed, we must show an error for previews on published post (remote auto-save) on Jetpack sites.

More information on the iOS ticket: https://github.com/wordpress-mobile/WordPress-iOS/issues/11684

### PR DESCRIPTION
**Pull request by maxme**
Fixes #10161: Hide previews for locally modified published posts on Jetpack connected sites

Case 1:
- On a Jetpack site
- Open a published post and modify its content
- Tap the "Preview" button in the overflow menu
- You should get a "Preview unavailable" screen

Case 2:
- On a Jetpack site
- Open a published post and modify its content
- Go back on the post list, tap "Preview" under the "More" button
- You should get a "Preview unavailable" screen

Case 3:
- On a Jetpack site
- Open a draft (or create a new one), modify its content
- Tap the "Preview" button in the overflow menu
- You should get the actual preview

Case 4: 
- On a WPCom site
- Open a published post and modify its content
- Tap the "Preview" button in the overflow menu
- You should get the actual preview

[Demo video](https://bia.is/s/HrWH/10161-preview-unavailable-jetpack-sites.webm)

Ping @osullivanchris - I wonder if we should show a specific message in that case (maybe have a look at the convo on p77Llu-cc4-p2 first).

### CODE REVIEW

**Comment by maxme** [comment]
Note: We should wait a week or so before merging that one, there is an upcoming API fix that might fix the underlying issue.

**Comment by jkmassel** [comment]
I'm bumping this to `13.1` as part of the `13.0` code freeze.

If you'd like it to be released as part of `13.0`, please merge it against `release/13.0` then DM me, and I'll be happy to issue a new beta release! :)

**Comment by maxme** [comment]
@malinajirka @shiki - Jetpack sites are still affected by the bug (a fix was deployed for simple sites, but it doesn't fix the issue for Atomic and Jetpack sites). I think we should merge that one and revert it when the final API fix is deployed.

**Comment by shiki** [comment]
@maxme Quick question on Case 3. I'm testing this on a Jurassic site. For a draft that's been uploaded to the server but with no changes, when I edit it:

- Without doing anything: Preview is unavailable
- Changing something: Preview is unavailable



Is this expected?

**Comment by maxme** [comment]
> I think I'm misunderstanding what a Jetpack site is. I'll need your help in testing this. 😬

I think your jurassic.ninja site is not connected to your wordpress.com account, and act like a self hosted site instead of a Jetpack connected site:

- A Jetpack site is a self hosted site we access via the wpcom REST API.
- `!site.isUsingWpComRestApi`, means a site we don't access via wpcom REST API (currently, the only other API we support is the self hosted XMLRPC API), so it's not a Jetpack site (or a wpcom site).


> we have `PostListAction.PreviewPost` and `PostListAction.ViewPost`. What is the difference between them?

The difference is that `ViewPost` will directly open the web view with the URL. `PreviewPost` will first try to upload or auto-save (depending on the post status) the post, wait for the server to reply, and then open the web view with the URL we got from server's response.


> For a draft that's been uploaded to the server but with no changes, when I edit it:
>    - Without doing anything: Preview is unavailable
>    - Changing something: Preview is unavailable

I'll double check self hosted cases. IIRC we should show the preview in these cases for _drafts_.

**Comment by maxme** [comment]
> > For a draft that's been uploaded to the server but with no changes, when I edit it:
> > 
> > * Without doing anything: Preview is unavailable
> > * Changing something: Preview is unavailable
> 
> I'll double check self hosted cases. IIRC we should show the preview in these cases for _drafts_.

After checking our discussion log, these 2 cases for wpcom / jetpack or self hosted sites should upload and preview drafts.

It's not related to this fix, I'll open another PR to fix it.

**Comment by maxme** [comment]
@shiki I added some useless tests to document the behavior (I'll add better tests for self hosted sites in the other fix).

File: WordPress/src/test/java/org/wordpress/android/ui/posts/RemotePreviewLogicHelperTest.kt [hunk 1]
```diff
@@ -237,4 +238,68 @@ class RemotePreviewLogicHelperTest {
...
+                activity,
+                WPWebViewUsageCategory.REMOTE_PREVIEW_NOT_AVAILABLE,
+                post.title
+        )
+    }
+
+    @Test
+    fun `preview available for Jetpack sites on draft with modification`() {
+        // Given
+        // next stub not used (made lenient) in case we update future logic.
```
Comment (shiki) [inline WordPress/src/test/java/org/wordpress/android/ui/posts/RemotePreviewLogicHelperTest.kt]: This comment and `lenient()` is 🎉

**Comment by maxme** [comment]
> Maybe they can be merged

**Comment by shiki** [comment]
> I wonder if we should change the UI as well, I'm not sure if the distinction between View/Preview is useful for users.

That's a good idea.

**Pull request merged**